src/main/java/org/generationcp/middleware/api/germplasm/GermplasmService.java

```diff
@@ -28,6 +28,10 @@
 
 public interface GermplasmService {
 
+	Germplasm getGermplasmByGID(Integer gid);
+
+	Germplasm saveGermplasm(Germplasm germplasm);
```

Comment:
Same here as much as possible let's not expose Germplasm entity, especially for saving. Since we are saving germplasm names, I think we should use GermplasmNamingService for saving names